### ISSUE TITLE
Nier Automata (524220) and other games crashing with Proton 5.0-9 but not with custom forks

### ISSUE THREAD

**Issue by CygnusTerminal** [ValveSoftware/Proton#4231]
Today i reinstalled Arch on my rebuilt PC, and trying to run a game, for example, NieR Automata, with Proton 5.0-9, crashes to desktop when loading. Custom forks seem to be working. I'll attach the log, never had this happen to me. I tried re-validating files, but they seemed to be fine. Haven't tested other official Proton versions.
[steam-524220.log](https://github.com/ValveSoftware/Proton/files/5302680/steam-524220.log)

**Comment by kisak-valve** [comment]
Hello @CygnusTerminal, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by CygnusTerminal** [comment]
Hello @kisak-valve, sorry i forgot, there you go: https://gist.github.com/CygnusTerminal/1a8c6c0b0a9ae0eca446c5a243d5c231

**Comment by CygnusTerminal** [comment]
Reinstalled Arch to no avail, games still crash under official proton builds.

**Comment by popsUlfr** [comment]
Just a shot in the dark but can you reproduce these crashes with the `linux-zen` kernel ? (it has the fsync patches)
I've had some weird behaviour with some apps using the tkg pds kernel in the past.

**Comment by CygnusTerminal** [comment]
> Just a shot in the dark but can you reproduce these crashes with the `linux-zen` kernel ? (it has the fsync patches)
> I've had some weird behaviour with some apps using the tkg pds kernel in the past.

I can confirm standard arch kernels and zen kernels aren't working too.

**Comment by noauthoritybuturself** [comment]
same thing happening to me with proton 5.0.9 and 4*. 
proton 3* seems to work fine, but the game is too slow. only playable at 950x540 resolution. my gpu (nvidia 965M) is supposed to work fine for this game.

**Comment by CygnusTerminal** [comment]
> 
> 
> same thing happening to me with proton 5.0.9 and 4*.
> proton 3* seems to work fine, but the game is too slow. only playable at 950x540 resolution. my gpu (nvidia 965M) is supposed to work fine for this game.

Are you using Arch or an Arch based distro? I also had a crash to desktop on Windows 10 after ~20 minutes of playing. At this rate, i think the game is misbehaving. But then again, i can't really confirm since i haven't played NieR again since thursday. About Proton 3*, game was platinum when proton 3.7 was around, so it may also be a problem with Arch Linux and a library.

**Comment by noauthoritybuturself** [comment]
> > same thing happening to me with proton 5.0.9 and 4*.
> > proton 3* seems to work fine, but the game is too slow. only playable at 950x540 resolution. my gpu (nvidia 965M) is supposed to work fine for this game.
> 
> Are you using Arch or an Arch based distro? I also had a crash to desktop on Windows 10 after ~20 minutes of playing. At this rate, i think the game is misbehaving. But then again, i can't really confirm since i haven't palyed NieR again since thursday. About Proton 3*, game was platinum when proton 3.7 was around, so it may also be a problem with Arch Linux and a library.

nop, i'm using linux mint 20 and nvidia 450 drivers. 
[my log](https://gist.github.com/noauthoritybuturself/f57b10ebd38e1b2559276e074f4b9a31)

**Comment by CygnusTerminal** [comment]
Oh, i see, then it has to be something related to the game/steam itself. I'm getting the same unhandled exception errors on Arch Linux with an AMD RX 480. I'll keep digging and i'll see what i can find. Thanks for chiming in!

**Comment by noauthoritybuturself** [comment]
thanks, i'll be checking this thread and report if i find something too :)

**Comment by noauthoritybuturself** [comment]
btw, careful with trying a bunch of different proton version, just got locked out of the game I just bought.. ridiculous.
https://support.codefusion.technology/nierautomata_9idut6rw/?e=88500006&l=english

**Comment by CygnusTerminal** [comment]
Yeah, that is Denuvo for you. It has a limit of machines you can install the game to, and since Wine prefixes are interpreted as new pcs, well, Denuvo locks you out for 24hs.

**Comment by noauthoritybuturself** [comment]
ok, so after 24h i was able to do a bit more testing.. it seems to work with Proton-5.9-GE-6-ST. not a fix, but something. still slow, but that's probably something related to my card i still need to figure out..

**Comment by CygnusTerminal** [comment]
Yes, that is what i observed too, game runs fine on my end with a custom proton fork. Game running slow probably is a driver issue, have you upgraded the driver? Can you downgrade if so? I was having problems with the latest nvidia driver on my gtx 960m (ugh) laptop.

**Comment by noauthoritybuturself** [comment]
*kinda* fixed it. apparently cinnamon was a big part of the problem. installed xfced, disabled compositor and it's running much better. 720p@60fps and 900p@30fps, medium settings, with occasional drops. that's probably the best I can expect with a 950m. it looks good.
weirdly, every time i change resolution or any other graphic setting in game, it slows it down to 10fps or worst, no matter which setting I use, so i have no choice than to exit the game and restart.

**Comment by CygnusTerminal** [comment]
Totally out of topic, have you tried the FAR fix? Supposedly works with proton-tkg (https://gist.github.com/MiyacoGBF/6fd49ae4a73a9a7f4d13c488bff2da77) Also, proton builds higher than 5.9 i think also crashes. I'll try downgrading wine system-wide, since it started malfuctioning with the latest wine builds.
EDIT: For clarification, and after testing, downgrading wine didn't work. I thought it would since it connected pretty well, the game stopped working after upstream wine updated to 5.17, and when i uninstalled the game via steam yesterday (to reinstall it, just to be safe) it tried to update my wine prefix (the one at .wine in the home folder). Still not understanding why the game suddenly stopped working with official proton builds.

**Comment by CygnusTerminal** [comment]
New update: Tested with mesa-git master to no avail. Game refuses to work with current official proton builds.

**Comment by Persephone-the-Sheep** [comment]
It might be a problem with the game because my laptop can't play the game with proton 4.11 and 5.0 or wine versions greater then 4.0  because they recreate a windows bug that crashes the game.

**Comment by CygnusTerminal** [comment]
> It might be a problem with the game because my laptop can't play the game with proton 4.11 and 5.0 or wine versions greater then 4.0 because they recreate a windows bug that crashes the game.

Might be, game was working fine until weeks ago, i can't debug it nor i can't fix it by myself, at this rate i'll just go back to Windows until proton gets rebased and see if that helps the game run fine again.

**Comment by Cxpher** [comment]
I have it running fine on Arch (all latest updates) with both 5.0.9 and GE.
